chainexchange/pubsub.go

```diff
@@ -38,18 +38,24 @@ type PubSubChainExchange struct {
 	pendingCacheAsWanted chan Message
 	topic                *pubsub.Topic
 	stop                 func() error
+	encoding             *encoding.ZSTD[*Message]
 }
 
 func NewPubSubChainExchange(o ...Option) (*PubSubChainExchange, error) {
 	opts, err := newOptions(o...)
 	if err != nil {
 		return nil, err
 	}
+	zstd, err := encoding.NewZSTD[*Message]()
```

Comment:
Are we going with ZSTD by default?

Reply:
For chain exchange yes. For GPBFT it's configurable via manifest.

Happy to make it configurable for chain exchange too if you think it's worth doing.